map_creator/src/find_base_byAverage.cpp

```diff
@@ -0,0 +1,358 @@
+//Sphere dataset
+     sphere_group = H5Gopen (file, "/Spheres", H5P_DEFAULT);
+     sphere_dataset = H5Dopen (sphere_group, "sphere_dataset", H5P_DEFAULT);
+     multimap<vector<double>, double > SphereCol;
+     hd5.h5ToMultiMapSpheres(sphere_dataset, SphereCol);
+
+//Resolution Attribute
+     float res;
+     attr = H5Aopen(sphere_dataset,"Resolution",H5P_DEFAULT);
+     herr_t ret = H5Aread(attr, H5T_NATIVE_FLOAT, &res);
+
+//Starting to create the Inverse Reachability map. The resolution will be the same as the reachability map
+   
+
+    SphereDiscretization sd;
+       vector<geometry_msgs::Pose> grasp_poses;
+    tf2::Quaternion grasp_quat1(0.5, 0.5, 0.5, 1); 
+    grasp_quat1.normalize();
+    tf2::Quaternion grasp_quat2(0, 0, 0, 1); 
+    grasp_quat2.normalize();
+    tf2::Quaternion grasp_quat3(0, 0.55, 0.28, 1); 
+    grasp_quat3.normalize();   
+
+
+
+
+    geometry_msgs::Pose grasp_pose1;
+    grasp_pose1.position.x = 0;
+    grasp_pose1.position.y = 0;
+    grasp_pose1.position.z = 1;
+    grasp_pose1.orientation.x = grasp_quat1[0];
+    grasp_pose1.orientation.y = grasp_quat1[1];
+    grasp_pose1.orientation.z = grasp_quat1[2];
+    grasp_pose1.orientation.w = grasp_quat1[3];
+    grasp_poses.push_back(grasp_pose1);   
+
+
+    geometry_msgs::Pose grasp_pose2;
+    grasp_pose2.position.x = 0.55;
+    grasp_pose2.position.y = 0;
+    grasp_pose2.position.z = 0.55;
+    grasp_pose2.orientation.x = grasp_quat2[0];
+    grasp_pose2.orientation.y = grasp_quat2[1];
+    grasp_pose2.orientation.z = grasp_quat2[2];
+    grasp_pose2.orientation.w = grasp_quat2[3];
+    grasp_poses.push_back(grasp_pose2);   
+
+    geometry_msgs::Pose grasp_pose3;
+    grasp_pose3.position.x = 0.99;
+    grasp_pose3.position.y = 0.23;
+    grasp_pose3.position.z = 0.34;
+    grasp_pose3.orientation.x = grasp_quat3[0];
+    grasp_pose3.orientation.y = grasp_quat3[1];
+    grasp_pose3.orientation.z = grasp_quat3[2];
+    grasp_pose3.orientation.w = grasp_quat3[3];
+    grasp_poses.push_back(grasp_pose3);
+
+    multiset<pair<double, vector<double> > > scoreWithSp;
+    for (map<vector<double>, double > ::iterator it = SphereCol.begin();it != SphereCol.end();++it)
+    {
+      scoreWithSp.insert(pair<double, vector<double> >(it->second, it->first ));
+    }
+
+    ROS_INFO("Numer of Spheres : %lu",scoreWithSp.size());
+    
+    vector<vector<double> > highScoreSp;
+    for (multiset<pair<double, vector<double> > > ::reverse_iterator it = scoreWithSp.rbegin();it != scoreWithSp.rend();++it)
+    {
+	 highScoreSp.push_back(it->second);
+
+    }
+
+    ROS_INFO("Size of high score sp: %lu", highScoreSp.size()); //filtered sp centers based on their scores
+    vector<geometry_msgs::Pose> probBasePoses;
+
+   int numofFinalPose = 10;
+
+    map_creator::WorkSpace ws;
+    for(int i=0;i<numofFinalPose;++i) //How many optimal base pose we need (But each pose is from a single sphere)
+    {
+      map_creator::WsSphere wss;
+      wss.point.x = highScoreSp[i][0];
```

Comment:
You need to check to make sure that `highScoreSp > numofFinalPose` and that `numofFinalPose > 0` and handle any special cases accordingly (print error statement, return cleanly).  Be sure to do this for all scoring methods.